src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java

```diff
@@ -653,63 +532,137 @@ private void processOldSolidEntryPoints(Tangle tangle, SnapshotProvider snapshot
-                    progressLogger.getCurrentStep() < progressLogger.getStepCount()) {
-
-                MilestoneViewModel currentMilestone = nextMilestone;
-                DAGHelper.get(tangle).traverseApprovees(
-                        currentMilestone.getHash(),
-                        currentTransaction -> currentTransaction.snapshotIndex() >= currentMilestone.index(),
-                        currentTransaction -> {
-                            if (isNotOrphaned(tangle, currentTransaction.getHash(), targetMilestone)) {
-                                addTailsToSolidEntryPoints(targetMilestone.index(), solidEntryPoints,
-                                        currentTransaction);
-                            }
-                        }
-                );
-
-                solidEntryPoints.put(currentMilestone.getHash(), targetMilestone.index());
-
-                nextMilestone = MilestoneViewModel.findClosestPrevMilestone(tangle, currentMilestone.index(),
-                        snapshotProvider.getInitialSnapshot().getIndex());
-
-                progressLogger.progress();
-            }
-
+            solidEntryPoints.putAll(getSolidEntryPoints(targetMilestone.index(), progressLogger));
             progressLogger.finish();
         } catch (Exception e) {
             progressLogger.abort(e);
-
             throw new SnapshotException("could not generate the solid entry points for " + targetMilestone, e);
         }
     }
+    
+    /**
+     * Generates entrypoints based on target index down to the maximum depth of the node
+     * 
+     * @param targetIndex The milestone index we target to generate entrypoints until.
+     * @param progressLogger The logger we use to write progress of entrypoint generation
+     * @return a map of entrypoints or <code>null</code> when we were interrupted
+     * @throws Exception When we fail to get entry points due to errors generally caused by db interaction
+     */
+    private Map<Hash, Integer> getSolidEntryPoints(int targetIndex, ProgressLogger progressLogger) throws Exception {
+        Map<Hash, Integer> solidEntryPoints = new HashMap<>();
+        solidEntryPoints.put(Hash.NULL_HASH, targetIndex);
+        log.info("Generating entrypoints for {}", targetIndex);
+        
+        int sepDepth = getSepDepth();
+        // Co back a but below the milestone. Limited to maxDepth or genisis
+        int startIndex = Math.max(snapshotProvider.getInitialSnapshot().getIndex(), targetIndex - sepDepth
+                ) + 1; // cant start at last snapshot now can we, could be 0!
+        
+        progressLogger.start(startIndex);
+        
+        // Iterate from a reasonable old milestone to the target index to check for solid entry points
+        for (int milestoneIndex = startIndex; milestoneIndex <= targetIndex; milestoneIndex++) {
+            if (Thread.currentThread().isInterrupted()) {
+                Thread.currentThread().interrupt();
+            }
```

Comment:
What you are doing here is setting the `interrupt` flag on...
But it was ready on and you didn't clear it now with `isInterrupted`... and you are not stopping the operation for sure...
So now this is completely broken

What I asked you before was either:
1. Do it properly with `InterruptedException`
2. Leave it like it was before but use `isInterrupted` instead of `interrupted` to not clear the flag. This would still be not the proper way to handle interrupts but at least it was less in your face bad than the original

So choose one of the above.. I'd read the links I sent you again and go with 1 just because I think you should probably know how interrupts in java work